docs/branching.md

```diff
@@ -0,0 +1,35 @@
+# Git Branching
+
+The following is a modified version of the gitflow branching strategy described in <https://nvie.com/posts/a-successful-git-branching-model/>
+
+## Main Branches
+
+- **master** - `origin/master` is the main branch where the source code of HEAD always reflects a production-ready state.
```

Comment:
nit: If we use `origin/master`, then we should also use `origin/release/X.Y.Z...`. 

Maybe it makes sense to drop the remote?

Reply:
i'm ambivalent here - it was an attempt at more verbosity, but i agree it would cause the extra burden of keeping that verbosity for all branches. 
i'm removing it

Reply:
696e78f4e79bd53482e5a6ebede67d46fad36d55